### PR TITLE
Fix: SIGTERM and any other signal gracefully shuts children process

### LINKED ISSUE #136: Process doesn't stop on SIGTERM
I've been using babel-watch for ages and always had issues with zombie processes appearing from time to time. It wasn't a big deal, as I simply cleared all the "node" threads in the Activity Monitor. From my experience the problem was usually caused by crashed express server.

Anyways, the time has come for me to cover my configuration with unit testing and so that problem with hanging processes became super apparent, it simply hangs every launch when triggered by an external child_process call (execa included).

Spent whole day trying to terminate the process from outside - no use, even terminating whole tree of processes doesn't work. And then I finally started debugging the issue from inside of babel-watch.

The problem was rather apparent, there is no code for graceful SIGTERM handling. The only processes signal is SIGINT (which fires from ctrl+c). But if process closes during another signal, it just hangs and becomes a zombie. Popular libraries use signal-exit to cover all the possible termination reasons.

Got a pull request, waiting for your feedback :)

Will test all the listed cases today/tomorrow and remove _Draft:_ prefix

### PR DESCRIPTION
**Pull request by IMalyugin**
Fixes #136

The code is working, but i still need to verify various use-cases:
- [x] Verify that works under execa
- [x] Verify that ctrl+c stops babel-watch
- [x] Verify that underlying app crash is handled properly
- [x] Verify auto restart after changing files works
- [x] Verify manual restart via "rs" works

### CODE REVIEW

File: babel-watch.js [hunk 1]
```diff
@@ -156,12 +157,13 @@ const watcher = chokidar.watch(program.watch, {
...
-process.on('SIGINT', function() {
-  debugInit('SIGINT caught, closing.');
+
+onExit(function(code, signal) {
+  debugInit(`${signal} caught, closing.`);
   watcher.close();
   killApp();
-  process.exit(0);
-});
+  process.exit(code || 0);
```
Comment (IMalyugin) [inline babel-watch.js:165]: Sending SIGINT, SIGTERM and possibly any other signal, gives `code: null`
Running process.exit(code), results into onExit called with the given code value

File: .gitignore [hunk 2]
```diff
@@ -34,4 +34,5 @@ node_modules
 
 # vscode
 # not sure if you want my settings committed to your repo
-.vscode
\ No newline at end of file
+.vscode
+.idea
```
Comment (IMalyugin) [inline .gitignore:38]: Not sure if you want my settings either :)

File: babel-watch.js [hunk 3]
```diff
@@ -156,12 +157,13 @@ const watcher = chokidar.watch(program.watch, {
 let watcherInitialized = (program.watch.length === 0);
 debugInit('Initializing babel-watch with options: %j', program.opts());
 
-process.on('SIGINT', function() {
-  debugInit('SIGINT caught, closing.');
+
+onExit(function(code, signal) {
+  debugInit(`${signal} caught, closing.`);
```
Comment (IMalyugin) [inline babel-watch.js]: Wonder if it should print a message if exited via process.exit. Might add signal
```
signal || `code ${code}`
```

**Comment by IMalyugin** [comment]
Okay, checked restart and crash - everything works perfectly.

Checked different SIGNALS - they do terminate child process.
As for codes - no extra messages printed during restart of the watcher. But If i call `process.exit(-1)` manually in the `babel-watch.js`, i get `exitCode -1 received, closing.` The only manual calls there are - are during some critical failure anyways, so it doesn't matter.

**Comment by STRML** [comment]
Excellent - high quality PR and issue description, thanks a ton!

**Comment by STRML** [comment]
Released in `v7.7.1`.

**Comment by STRML** [comment]
Just curious, any reason to use a 7-year-old version of signal-exit rather than the latest? I don't see any major applicable changes (although blocking SIGPROF might be useful), but struck me as odd.

**Comment by IMalyugin** [comment]
> Just curious, any reason to use a 7-year-old version of signal-exit rather than the latest? I don't see any major applicable changes (although blocking SIGPROF might be useful), but struck me as odd.

You're absolutely right, no excuse. Thought I used 2.1.2 myself, but nope, using 3.x.x. That's just my bad habit of adding packages manually to package.json via IDe suggestions. Especially after I had to downgrade a whole bunch of dependencies recently due to ESM.

Got a new PR out for a bump

**Pull request merged**